File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        curr=0
        mx=0
        for i in range(0,k-1):
            if(s[i] in "aeiou") :
                curr+=1
        mx=curr
        for i in range(k-1,len(s)):
            if(s[i] in "aeiou"):
                curr+=1
                mx=max(curr,mx)
            if(s[i+1-k] in "aeiou"):
                curr-=1

        return mx
        

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
public:
    int maxVowels(string s, int k) {
        int curr=0;
        int mx=0;
        string p = "aeiou";
        for(int i = 0 ;i<k-1;i++){
            if(p.contains(s[i])){
                curr++;
            }
        }
        mx=curr;
        for(int i =k-1;i<s.size();i++){
            if(p.contains(s[i])){
                curr++;
                mx=max(mx,curr);
            }
            if(p.contains(s[i+1-k])){
                curr--;
            }
        }
        return mx;
    }
};